sophus/rxso2.hpp

```diff
@@ -503,26 +495,20 @@ class RxSO2 : public RxSO2Base<RxSO2<Scalar_, Options>> {
   ///
   SOPHUS_FUNC static Sophus::Matrix<Scalar, num_parameters, DoF> Dx_exp_x(
       Tangent const& a) {
-    using std::cos;
     using std::exp;
-    using std::sin;
     Scalar const theta = a[0];
     Scalar const sigma = a[1];
 
-    Sophus::Matrix<Scalar, num_parameters, DoF> J;
-    J << -sin(theta), cos(theta), cos(theta), sin(theta);
-    return J * exp(sigma);
+    return SO2<Scalar>::exp(theta).matrix().rowwise().reverse() * exp(sigma);
   }
 
   /// Returns derivative of exp(x) wrt. x_i at x=0.
   ///
   SOPHUS_FUNC static Sophus::Matrix<Scalar, num_parameters, DoF>
   Dx_exp_x_at_0() {
-    Sophus::Matrix<Scalar, num_parameters, DoF> J;
-    static Scalar const i(1.);
-    static Scalar const o(0.);
-    J << o, i, i, o;
-    return J;
+    return Sophus::Matrix<Scalar, num_parameters, DoF>::Identity()
```

Comment:
That seems a little overkill for creating such a matrix, but I guess it can deduse Identity().rowwise().reverse() at compile time...